File: yixiang_be/src/main/java/com/tongji/settings/service/SettingsService.java
package com.tongji.settings.service;

import com.tongji.settings.api.dto.SettingsPatchRequest;
import com.tongji.settings.api.dto.SettingsResponse;

public interface SettingsService {
    SettingsResponse get(long userId);
    SettingsResponse patch(long userId, SettingsPatchRequest req);
}

File: yixiang-web/src/services/settingsService.ts
import { apiFetch } from '@/lib/apiClient';

export interface SettingsResponse {
  notificationPref: Record<string, boolean>;
  privacy: Record<string, unknown>;
  theme: string;
  language: string;
}

export const settingsService = {
  get: () => apiFetch<SettingsResponse>('/api/v1/settings'),
  patch: (body: Partial<SettingsResponse>) =>
    apiFetch<SettingsResponse>('/api/v1/settings', { method: 'PATCH', body: JSON.stringify(body) }),
};
